File: atividade01/3.cpp
invlude 


Conta contas[] = [];

class Conta {
public:
    double numero;
    double saldo;
}




function criarConta(numero: number, saldoInicial: number): void {
    contas.push({ numero, saldo: saldoInicial });
    console.log(`Conta ${numero} criada com saldo inicial de R$${saldoInicial}.`);
}

function depositar(numeroConta: number, valor: number): void {
    const conta = contas.find(c => c.numero === numeroConta);
    if (conta) {
        conta.saldo += valor;
        console.log(`Depósito de R$${valor} realizado na conta ${numeroConta}. Novo saldo: R$${conta.saldo}.`);
    } else {
        console.log("Conta não encontrada.");
    }
}

function sacar(numeroConta: number, valor: number): void {
    const conta = contas.find(c => c.numero === numeroConta);
    if (conta) {
        if (conta.saldo >= valor) {
            conta.saldo -= valor;
            console.log(`Saque de R$${valor} realizado na conta ${numeroConta}. Novo saldo: R$${conta.saldo}.`);
        } else {
            console.log("Saldo insuficiente.");
        }
    } else {
        console.log("Conta não encontrada.");
    }
}

function consultarSaldo(numeroConta: number): void {
    const conta = contas.find(c => c.numero === numeroConta);
    if (conta) {
        console.log(`Saldo da conta ${numeroConta}: R$${conta.saldo}.`);
    } else {
        console.log("Conta não encontrada.");
    }
}

criarConta(123, 1000);
depositar(123, 500);
sacar(123, 200);
consultarSaldo(123);

File: atividade02/3.ts
abstract class ContaBancaria {
    protected numero: number;
    protected saldo: number;

    constructor(numero: number, saldo: number) {
        this.numero = numero;
        this.saldo = saldo;
    }

    depositar(valor: number) {
        this.saldo += valor
    }

    sacar(valor: number): boolean {
        if (this.saldo < valor){
            return false;
        }
        this.saldo -= valor;
        return true;
    } 
}

class ContaCorrente extends ContaBancaria {
    private limiteDeChequeEspecial: number;

    constructor(numero: number, saldo: number, limiteDeChequeEspecial: number) {
        super(numero, saldo);
        this.limiteDeChequeEspecial = limiteDeChequeEspecial;
    }

    sacar(valor: number): boolean {
        if (this.saldo + this.limiteDeChequeEspecial < valor){
            return false;
        }
        this.saldo -= valor;
        return true;
    }

}

class ContaPoupanca extends ContaBancaria {
    private taxaJuros: number
    constructor(numero: number, saldo: number, taxaJuros: number) {
        super(numero, saldo);
        this.taxaJuros = taxaJuros;
    }

    aplicarJuros() {
        this.saldo += this.taxaJuros * this.saldo
    }
}

let cc = new ContaCorrente(1, 50, 50);
cc.depositar(100);
console.log(cc.sacar(100));
console.log(cc.sacar(150));

let cp = new ContaPoupanca(2, 50, 5);
cp.aplicarJuros()